Automattic/wp-calypso
# Update/purchase tracks component

Replaces `recordPageView` with new tracks component (cherry-picked addition from #24861)

https://github.com/Automattic/wp-calypso/blob/dab66765c342f27002cf8be02d53221643170e07/client/me/purchases/utils.js#L26-L50

Event names become literal strings according to best practices. Previously they were ``` `calypso_${ trackingSlug }_purchase_view` ```

Here's the translation:

```jsx
recordPageView( 'cancel_private_registration', this.props, nextProps )
// becomes the component component
<TrackPurchasePageView
				eventName="calypso_cancel_private_registration_purchase_view"
				purchaseId={ this.props.purchaseId }
/>
```

## Testing
1. Verify event names have been correctly translated.
1. `localStorage.debug = 'calypso:analytics:tracks'`
1. Browse around the different purchase views and verify that the tracks events are fired as expected.
